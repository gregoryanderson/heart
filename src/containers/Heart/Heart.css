main {
  background-image: linear-gradient(#ede7f6,#e8eaf6);
}

.loading {
  border: none;
  height: 300px;
}

.loading-div {
  text-align: center;
  padding-bottom: 500px;
}

.modal {
  display: flex;
  justify-content: center;
  height: 300px;
  width: 300px;
  background-color: pink;
  margin: auto;
  margin-top: 50px
}