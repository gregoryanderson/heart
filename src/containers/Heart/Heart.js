import React, { Component } from "react";
import "./Heart.css";
import { getPaintingsFromApiCalls } from "../../apiCalls/apiCalls.js";
import { establishPaintingsInRedux, addFavoriteInRedux, deleteFavoriteInRedux} from '../../actions'
import { Route, Redirect } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import Wall from "../Wall/Wall";
import Nav from "../Nav/Nav";

class Heart extends Component {
  constructor(props) {
    super(props);
    this.state = {
      paintings: [],
      error: ""
    };
  }

  async componentDidMount() {
    try {
      const pieces = await getPaintingsFromApiCalls();
      this.props.establishPaintingsInRedux(pieces)
    } catch ({ message }) {
      this.setState({ error: message });
    }
  }


  handleFavorite = (artId) => {
    const foundPainting = this.props.paintings.find(painting => {
      return painting.id == artId
    })
    if (!foundPainting.isFav){
      foundPainting.isFav = !foundPainting.isFav
      this.props.addFavoriteInRedux(foundPainting)
    } else {
      foundPainting.isFav = !foundPainting.isFav
      this.props.deleteFavoriteInRedux(foundPainting)      
    }
  }


  render() {
    return (
      <section>
        <Nav />
        {!!this.props.paintings.length && <Wall paintings={this.props.paintings} handleFavorite={this.handleFavorite}/>}
      </section>
    );
  }
}

const mapStateToProps = state => ({
  paintings: state.paintings,
  favorites: state.favorites
});

const mapDispatchToProps = dispatch => ({
  establishPaintingsInRedux: paintings => dispatch(establishPaintingsInRedux(paintings)),
  addFavoriteInRedux: favorite => dispatch(addFavoriteInRedux(favorite)),
  deleteFavoriteInRedux: favorite => dispatch(deleteFavoriteInRedux(favorite))
})



export default connect(mapStateToProps, mapDispatchToProps)(Heart);

Heart.propTypes = {};
