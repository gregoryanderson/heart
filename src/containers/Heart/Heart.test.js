import React from "react";
import { shallow, mount } from "enzyme";
import { Heart, mapStateToProps } from "./Heart";
import { MemoryRouter } from "react-router";

describe("Heart", () => {
  let wrapper, mockHandleFavorite, mockHandleSearch, mockFavorites, mockFacets, mockPaintings;

  beforeEach(() => {
    const mockHandleFavorite = jest.fn();
    const mockHandleSearch = jest.fn();

    const mockFavorites = [
      {
        title: "Cupboard",
        id: "en-BK-1975-81",
        artist: "Herman Doomer",
        width: 2026,
        height: 2500,
        url:
          "https://lh3.googleusercontent.com/tGI4dOAfJLBbewwspzXpUnSZxEKFACv9Y3FHqAxQUtN2p4AXt2MS9oFv6eJyIBtr7gvzmv58vSitMFVeHY0TGsfOfDN2=s0",
        isFav: false
      }
    ];
    const mockFacets = [
      {
        0: {}
      }
    ];
    const mockPaintings = [
      {
        title: "Clock and gunpowder horn",
        id: "en-NG-NM-7687",
        artist: "anonymous",
        width: 1672,
        height: 2500,
        url:
          "https://lh3.ggpht.com/lAJ1wnr_hEOncOfh9eKzvaS8w-fhLLq5yGlzHBctnjgyOzsbuP4cGIqP4q0A-YvnyXBhJi96il6NIZNhRVW-BVg2lW0=s0",
        isFav: false
      }
    ];

    wrapper = shallow(
      <Heart
        handleSearch={mockHandleSearch}
        handleFavorite={mockHandleFavorite}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("mapStateToProps", () => {
  it('should properly map to props', () => {
    const mockFavorites = [
      {
        title: "Cupboard",
        id: "en-BK-1975-81",
        artist: "Herman Doomer",
        width: 2026,
        height: 2500,
        url:
          "https://lh3.googleusercontent.com/tGI4dOAfJLBbewwspzXpUnSZxEKFACv9Y3FHqAxQUtN2p4AXt2MS9oFv6eJyIBtr7gvzmv58vSitMFVeHY0TGsfOfDN2=s0",
        isFav: false
      }
    ];
    const mockFacets = [
      {
        0: {}
      }
    ];
    const mockPaintings = [
      {
        title: "Clock and gunpowder horn",
        id: "en-NG-NM-7687",
        artist: "anonymous",
        width: 1672,
        height: 2500,
        url:
          "https://lh3.ggpht.com/lAJ1wnr_hEOncOfh9eKzvaS8w-fhLLq5yGlzHBctnjgyOzsbuP4cGIqP4q0A-YvnyXBhJi96il6NIZNhRVW-BVg2lW0=s0",
        isFav: false
      }
    ];

    const mockState = {
      favorites: [
        {
          title: "Cupboard",
          id: "en-BK-1975-81",
          artist: "Herman Doomer",
          width: 2026,
          height: 2500,
          url:
            "https://lh3.googleusercontent.com/tGI4dOAfJLBbewwspzXpUnSZxEKFACv9Y3FHqAxQUtN2p4AXt2MS9oFv6eJyIBtr7gvzmv58vSitMFVeHY0TGsfOfDN2=s0",
          isFav: false
        }
      ],
      facets: [
        {
          0: {}
        }
      ],
      paintings: [
        {
          title: "Clock and gunpowder horn",
          id: "en-NG-NM-7687",
          artist: "anonymous",
          width: 1672,
          height: 2500,
          url:
            "https://lh3.ggpht.com/lAJ1wnr_hEOncOfh9eKzvaS8w-fhLLq5yGlzHBctnjgyOzsbuP4cGIqP4q0A-YvnyXBhJi96il6NIZNhRVW-BVg2lW0=s0",
          isFav: false
        }
      ]
    };
  
    expect(mapStateToProps(mockState).paintings).toEqual(mockPaintings);
    expect(mapStateToProps(mockState).facets).toEqual(mockFacets);
    expect(mapStateToProps(mockState).favorites).toEqual(mockFavorites);
  })
});

describe("Heart Routes", () => {
  let wrapper;

  it.skip("will go to the favorites page when favorites is called", () => {
    let mockPaintings = [
      {
        title: "Cupboard",
        id: "en-BK-1975-81",
        artist: "Herman Doomer",
        width: 2026,
        height: 2500,
        url:
          "https://lh3.googleusercontent.com/tGI4dOAfJLBbewwspzXpUnSZxEKFACv9Y3FHqAxQUtN2p4AXt2MS9oFv6eJyIBtr7gvzmv58vSitMFVeHY0TGsfOfDN2=s0",
        isFav: false
      }
    ];

    wrapper = mount(
      <MemoryRouter initialEntries={["/favorites"]}>
        <Heart paintings={mockPaintings} />
      </MemoryRouter>
    );
    expect(wrapper.find(Favorites)).toHaveLength(1);
    expect(wrapper.find(Album)).toHaveLength(0);
    expect(wrapper.find(CreateUserForm)).toHaveLength(0);
    expect(wrapper.find(FavoritesContainer)).toHaveLength(0);
    expect(wrapper.find(WelcomeContainer)).toHaveLength(0);
  });
});
