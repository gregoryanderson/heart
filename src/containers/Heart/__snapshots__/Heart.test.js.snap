// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heart should match the snapshot 1`] = `
<main>
  <section>
    <Nav
      handleSearch={[Function]}
    />
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/favorites"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/artist"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/artist/:artist"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/medium"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/medium/:medium"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/color"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/color/:color"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/century"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/century/:century"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/type"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/type/:type"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/place"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/place/:place"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/technique"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/technique/:technique"
      render={[Function]}
    />
    <Route
      path="/paintings/:id"
      render={[Function]}
    />
  </section>
</main>
`;
